File: cpp/08_classes.cpp
#include <iostream>
#include <string>
using namespace std;

class Person {
public:
    string name;
    int age;

    Person(string n, int a) : name(n), age(a) {}

    string greet() {
        return "hi i am " + name + " and i am " + to_string(age);
    }
};

int main() {
    Person me("ulas", 20);
    cout << me.greet() << endl;
    cout << me.name << endl;
    return 0;
}

File: rust/08_classes.rs
struct Person {
    name: String,
    age: u32,
}

impl Person {
    fn new(name: &str, age: u32) -> Self {
        Person {
            name: name.to_string(),
            age,
        }
    }

    fn greet(&self) -> String {
        format!("hi i am {} and i am {}", self.name, self.age)
    }
}

fn main() {
    let me = Person::new("ulas", 20);
    println!("{}", me.greet());
    println!("{}", me.name);
}
